### ISSUE TITLE
OCP4 notes NotReady after long outage

### ISSUE THREAD

**Issue by DanielFroehlich** [stormshift/support#89]
OCP4 has been offline for a couple of weeks due to RHV networking issues.
It's back online, but two nodes dont get ready:
```
[root@ocp4support ~]# oc get nodes
NAME                                           STATUS     ROLES    AGE      VERSION
compute-0.ocp4.stormshift.coe.muc.redhat.com   NotReady   worker   2y162d   v1.21.8+ee73ea2
compute-1.ocp4.stormshift.coe.muc.redhat.com   Ready      worker   2y162d   v1.21.8+ee73ea2
compute-2.ocp4.stormshift.coe.muc.redhat.com   Ready      worker   627d     v1.21.8+ee73ea2
control-0.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y162d   v1.21.8+ee73ea2
control-1.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y162d   v1.21.8+ee73ea2
control-2.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y162d   v1.21.8+ee73ea2
gpu.ocp4.stormshift.coe.muc.redhat.com         NotReady   worker   233d     v1.21.8+ee73ea2
```
could be expired certs, but not CSRs visible. IIRC, we had this already in the past, please investigate and fix.

**Comment by Javatar81** [comment]
```
journalctl -b -f -u kubelet.service -u crio.service

Jun 29 09:16:21 compute-0.ocp4.stormshift.coe.muc.redhat.com hyperkube[127013]: I0629 09:16:21.474377  127013 csi_plugin.go:1031] Failed to contact API server when waiting for CSINode publishing: csinodes.storage.k8s.io "compute-0.ocp4.stormshift.coe.muc.redhat.com" is forbidden: User "system:anonymous" cannot get resource "csinodes" in API group "storage.k8s.io" at the cluster scope
Jun 29 09:16:21 compute-0.ocp4.stormshift.coe.muc.redhat.com hyperkube[127013]: I0629 09:16:21.517961  127013 certificate_manager.go:556] Certificate expiration is 2022-07-02 08:22:21 +0000 UTC, rotation deadline is 2022-06-26 16:18:33.656373552 +0000 UTC
```

Certificates expired

**Comment by Javatar81** [comment]
After restoring certificates as described here https://github.com/stormshift/support/issues/72

```
oc get nodes
NAME                                           STATUS     ROLES    AGE      VERSION
compute-0.ocp4.stormshift.coe.muc.redhat.com   Ready      worker   2y163d   v1.21.8+ee73ea2
compute-1.ocp4.stormshift.coe.muc.redhat.com   Ready      worker   2y163d   v1.21.8+ee73ea2
compute-2.ocp4.stormshift.coe.muc.redhat.com   Ready      worker   627d     v1.21.8+ee73ea2
control-0.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y163d   v1.21.8+ee73ea2
control-1.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y163d   v1.21.8+ee73ea2
control-2.ocp4.stormshift.coe.muc.redhat.com   Ready      master   2y163d   v1.21.8+ee73ea2
gpu.ocp4.stormshift.coe.muc.redhat.com         NotReady   worker   234d     v1.21.8+ee73ea2
```

**Comment by Javatar81** [comment]
Removed gpu node because it cannot be recovered

**Comment by Javatar81** [comment]
Problem with degraded node in machine config pool (mcp)

`oc get mcp                                                                                                                       
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-152a60c24bdf3a8ecbacd81c68e0ab97   True      False      False      3              3                   3                     0                      2y163d
worker   rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3   False     True       True       3              0                   0                     1                      2y163d`

`oc describe mcp worker
Node compute-0.ocp4.stormshift.coe.muc.redhat.com is reporting: "getting pending state from journal: invalid character 'J' looking for beginning of value"
`

Identified similar issue in knowledge base: https://access.redhat.com/solutions/5517491

**Comment by Javatar81** [comment]
As proposed by knowledge base article:
```
journalctl --verify
Journal file /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e281194540ab-e13e69b84322006e.journal~ is truncated, ignoring file.
7fffde8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d54a2fd-0005e289a81f8a0d.journal                       
7fffdc8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d5ac075-0005e2900e05883e.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000@0005e29431d98e89-65701ac6bbd90292.journal~                                                    
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000850000.journal                                                                                 
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000750000.journal                                                                                 
7fffcf8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@848c8bd6aa904aec95ea09e3731817de-0000000001de2f75-0005df5fa68a32b0.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-0000000000000001-0005e05da0cd05c9.journal                       
7fffe48: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d4e0d4e-0005e0741ee7e52d.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-0000000000079ee0-0005e1155f3076d6.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d4e8566-0005e2833e7e199a.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e05d8ca6553b-3a4ecfd4e5bb0d82.journal~                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-0000000000021cd4-0005e06081ea4f6e.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-0000000000087e75-0005e068ec586cce.journal                       
7fffe48: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-0000000000000001-0005e28118a8975a.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system.journal                                                                                          
7fffef8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-00000000000cbf80-0005e06e87da2b48.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-00000000000edfe5-0005e07151c54ff7.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d51995e-0005e07d65be5aca.journal                       
7fffe90: Unused data (entry_offset==0)                                                                                                                          
7fffee8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d55274b-0005e086e65427b1.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-0000000000018430-0005e0e903993b84.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e2942fc267ac-f606d45cf7cf94a9.journal~                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-0000000000043d68-0005e063502a37d6.journal                       
7fffe20: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d4fd506-0005e078f49b00d1.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d5360ff-0005e08236b60baf.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000720000.journal                                                                                 
7ffff50: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d56eefa-0005e08b43d5e403.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-000000004d58d0c4-0005e0909574a88a.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-0000000000000001-0005e0de3dde3875.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-00000000000491c3-0005e0ff3900801c.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-0000000000061854-0005e10a4c1833b1.journal                       
7fffe10: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d508f5b-0005e285624c659e.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d5298ff-0005e287851e20ac.journal                       
7fffdc8: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d56ac8c-0005e28bca360caf.journal                       
7ff9160: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@df6acc90f2964d80949c246119221f6c-000000004d58b68a-0005e28dec0c525a.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e05da122cbde-23168a8a467ffd2c.journal~                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000.journal                                                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000830000.journal                                                                                 
7fffe28: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-0000000000065df4-0005e0661e4cbc5a.journal                       
7fffe58: Unused data (entry_offset==0)                                                                                                                          
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@066f6ade849c4d96bc1a54ff30c2e790-00000000000a9ef9-0005e06bba61b676.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e0d9003a1cfa-87b48a6d307d13b8.journal~                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@0005e0de3e29d95f-34435dfec4121e92.journal~                                                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/system@6e77e1acce954e64af6d77c48151849a-0000000000030b2d-0005e0f425e65bb7.journal                       
PASS: /var/log/journal/dad33f20457b4dd99eae44cc64549002/user-1000@fac57c2d10b743efbdf819ef549b7c40-000000004d4cc32c-0005e28166ab609b.journal   
```

Deleted truncated file.

**Comment by Javatar81** [comment]
Waiting for mcp to recover from degraded state.  Mcp still degraded.

**Comment by Javatar81** [comment]
Image registry degraded => ImagePrunerDegraded: Job has reached the specified backoff limit

Knowledge Base: https://access.redhat.com/solutions/5370391

Solution successful. Registry available again.

**Comment by Javatar81** [comment]
Trying update 4.8.35 -> 4.8.43

**Comment by Javatar81** [comment]
Stuck during update:

```
oc get mcp --watch                                                                                                              
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-e902171eea97aefd62c4933756941299   True      False      False      3              3                   3                     0                      2y163d
worker   rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3   False     True       True       3              0                   0                     1                      2y163d
```

**Comment by Javatar81** [comment]
The machine config daemon on compute does not allow to view logs.

```
oc logs machine-config-daemon-rqrnt -c machine-config-daemon
error: You must be logged in to the server (the server has asked for the client to provide credentials ( pods/log machine-config-daemon-rqrnt))
```

Daemon pods on other nodes are all viewable.

**Comment by rbo** [comment]
```
 oc get node/compute-0.ocp4.stormshift.coe.muc.redhat.com  -o yaml | head -25
apiVersion: v1
kind: Node
metadata:
  annotations:
    csi.volume.kubernetes.io/nodeid: '{"openshift-storage.cephfs.csi.ceph.com":"compute-0.ocp4.stormshift.coe.muc.redhat.com","openshift-storage.rbd.csi.ceph.com":"compute-0.ocp4.stormshift.coe.muc.redhat.com"}'
    machineconfiguration.openshift.io/controlPlaneTopology: HighlyAvailable
    machineconfiguration.openshift.io/currentConfig: rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3
    machineconfiguration.openshift.io/desiredConfig: rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3
    machineconfiguration.openshift.io/reason: 'unexpected on-disk state validating
      against rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3: content mismatch for
      file "/etc/kubernetes/ca.crt"'
 
```

It looks like the `/etc/kubernetes/ca.crt` is changed and this is not expected from MCO.

**Comment by rbo** [comment]
Let's drain the node: `oc adm drain --delete-emptydir-data --ignore-daemonsets node/compute-0.ocp4.stormshift.coe.muc.redhat.com`

**Comment by rbo** [comment]
Let's force and rejoin the node:
```
systemctl stop kubelet
touch /run/machine-config-daemon-force
rm -rf /var/lib/kubelet/pki /var/lib/kubelet/kubeconfig
systemctl start kubelet
```

Doesn't help.... Let's copy the /etc/kubernetes/ca.crt from another host.
```
[root@ocp4bastion ~]# scp core@172.16.10.21:/etc/kubernetes/ca.crt .
ca.crt                                             100% 1123   865.4KB/s   00:00    
[root@ocp4bastion ~]# scp ca.crt core@@172.16.10.20:~/
Connection closed by 172.16.10.20 port 22
lost connection
[root@ocp4bastion ~]# scp ca.crt core@172.16.10.20:~/
ca.crt                                             100% 1123   966.8KB/s   00:00    
[root@ocp4bastion ~]# 
[core@compute-0 ~]$ mv ca.crt /etc/kubernetes/ca.crt 
mv: replace '/etc/kubernetes/ca.crt', overriding mode 0644 (rw-r--r--)? y
mv: inter-device move failed: 'ca.crt' to '/etc/kubernetes/ca.crt'; unable to remove target: Permission denied
[core@compute-0 ~]$ sudo mv ca.crt /etc/kubernetes/ca.crt 
[core@compute-0 ~]$ sudo reboot
```

**Comment by rbo** [comment]
Fixed
```
$ oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-e902171eea97aefd62c4933756941299   True      False      False      3              3                   3                     0                      2y163d
worker   rendered-worker-e4dadfe87d9d34e80d7b0c9fd65fcdf3   False     True       False      3              0                   0                     0                      2y163d
```
